binding/go/cheetah.go

```diff
@@ -0,0 +1,334 @@
+// Copyright 2022 Picovoice Inc.
+//
+// You may not use this file except in compliance with the license. A copy of the license is
+// located in the "LICENSE" file accompanying this source.
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the
+// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Go binding for Cheetah Speech-to-Text engine.
+
+package cheetah
+
+import (
+	"C"
+	"embed"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+//go:embed embedded
+var embeddedFS embed.FS
+
+// PvStatus type
+type PvStatus int
+
+// Possible status return codes from the Cheetah library
+const (
+	SUCCESS                  PvStatus = 0
+	OUT_OF_MEMORY            PvStatus = 1
+	IO_ERROR                 PvStatus = 2
+	INVALID_ARGUMENT         PvStatus = 3
+	STOP_ITERATION           PvStatus = 4
+	KEY_ERROR                PvStatus = 5
+	INVALID_STATE            PvStatus = 6
+	RUNTIME_ERROR            PvStatus = 7
+	ACTIVATION_ERROR         PvStatus = 8
+	ACTIVATION_LIMIT_REACHED PvStatus = 9
+	ACTIVATION_THROTTLED     PvStatus = 10
+	ACTIVATION_REFUSED       PvStatus = 11
+)
+
+type CheetahError struct {
+	StatusCode PvStatus
+	Message    string
+}
+
+func (e *CheetahError) Error() string {
+	return fmt.Sprintf("%s: %s", pvStatusToString(e.StatusCode), e.Message)
+}
+
+// Cheetah struct
+type Cheetah struct {
+	// handle for cheetah instance in C
+	handle uintptr
+
+	// AccessKey obtained from Picovoice Console (https://console.picovoice.ai/).
+	AccessKey string
+
+	// Absolute path to the file containing model parameters.
+	ModelPath string
+
+	// Duration of endpoint in seconds. A speech endpoint is detected when there is a
+	// chunk of audio (with a duration specified herein) after an utterance without any speech in it. Set to 0
+	// to disable endpoint detection.
+	EndpointDuration float32
+}
+
+type nativeCheetahInterface interface {
+	nativeInit(*Cheetah)
+	nativeProcess(*Cheetah, []int)
+	nativeFlush(*Cheetah, string)
+	nativeDelete(*Cheetah)
+	nativeSampleRate()
+	nativeVersion()
+}
+type nativeCheetahType struct{}
+
+// private vars
+var (
+	osName, cpu   = getOS()
+	extractionDir = filepath.Join(os.TempDir(), "cheetah")
+
+	defaultModelFile = extractDefaultModel()
+	libName          = extractLib()
+	nativeCheetah    = nativeCheetahType{}
+)
+
+var (
+	// Number of audio samples per frame.
+	FrameLength = nativeCheetah.nativeFrameLength()
+
+	// Audio sample rate accepted by Picovoice.
+	SampleRate = nativeCheetah.nativeSampleRate()
+
+	// Cheetah version
+	Version = nativeCheetah.nativeVersion()
+)
+
+// Init function for Cheetah. Must be called before attempting process
+func (cheetah *Cheetah) Init() error {
```

Comment:
Could be worth it to have a default constructor (`NewCheetah`), similar to the way we do in Rhino. That way they don't have to set endpoint duration, we can just set it to a default.